### PR TITLE
chore(test): standardize pytestmark usage across all test files

### LINKED ISSUE #195: Story 25.1: Standardize pytestmark Usage
## Story

As a **docvet contributor**,
I want pytestmark usage to be consistent across all unit test files,
so that test organization follows a clear convention and marker-based filtering works reliably.

## Acceptance Criteria

1. All 15 unit test files have `pytestmark = pytest.mark.unit` and all 2 integration test files have `pytestmark = pytest.mark.integration`
2. `uv run pytest` passes with all 737+ tests green
3. `.github/instructions/pytest.instructions.md` includes the convention and rationale
4. `pytest -m unit` and `pytest -m integration` usage is documented as local development shortcuts

---

Story file: `_bmad-output/implementation-artifacts/25-1-standardize-pytestmark-usage.md`
Branch: `feat/test-25-1-standardize-pytestmark-usage`

### PR DESCRIPTION
**Pull request by Alberto-Codes**
Previously 5 of 15 unit test files and 1 of 2 integration test files had
module-level `pytestmark` assignments, while the rest relied on class-level
decorators or had no markers at all. This made `pytest -m unit` and
`pytest -m integration` unreliable for filtering.

- Add `pytestmark = pytest.mark.unit` to 10 unit test files missing it
- Add `pytestmark = pytest.mark.integration` to `tests/integration/test_griffe_compat.py`
- Remove 8 redundant `@pytest.mark.unit` class-level decorators from `test_exports.py`
- Document the convention in `pytest.instructions.md` and add marker-based filtering to `CLAUDE.md`
- Standardize `importorskip` call in integration test (replace `__import__` workaround)

Test: `uv run pytest -m unit` (933 collected), `uv run pytest -m integration` (38 collected), `uv run pytest` (971 passed)

Closes #195

fix(bmad): add feature branch creation step to dev-story workflow

---

### Checklist
- [x] Self-reviewed my code
- [x] Tests pass (`uv run pytest`)
- [x] Lint passes (`uv run ruff check .`)
- [x] Types pass (`uv run ty check`)
- [ ] Breaking changes use `!` in title and `BREAKING CHANGE:` in body

### Review Focus
Convention consistency — verify all 17 test files have correct `pytestmark` assignments and no redundant class/function-level markers remain.

### Related
- Story 25.1: `_bmad-output/implementation-artifacts/25-1-standardize-pytestmark-usage.md`
- Original issue: #181 (file inventory)

### CODE REVIEW

File: tests/unit/test_lsp.py [hunk 1]
```diff
@@ -24,6 +24,8 @@
     did_save,
 )
 
+pytestmark = pytest.mark.unit
+
```
[the remark this answers is not in the document]
Comment (Alberto-Codes) [reply tests/unit/test_lsp.py]: Keeping as-is — `importorskip("pygls")` gates a cascade of conditional imports (`from lsprotocol`, `from docvet.lsp`) that follow with `noqa: E402`. Moving `pytestmark` before `importorskip` would place it in the middle of the import block, which is worse for readability. The convention treats `importorskip` as part of the import block, so `pytestmark` goes after the complete block. Updated the docs to clarify this edge case.

File: .github/instructions/pytest.instructions.md [hunk 2]
```diff
@@ -132,12 +132,40 @@ def test_invalid_config_raises_error():
...
 
+### Module-Level Markers (Convention)
+
+Every test file **MUST** have a module-level `pytestmark` assignment after imports (excluding `conftest.py` files, which contain fixtures, not tests):
+
+- Unit test files (`tests/unit/`): `pytestmark = pytest.mark.unit`
+- Integration test files (`tests/integration/`): `pytestmark = pytest.mark.integration`
+
+Place the assignment after all imports, before any test code:
+
```
[the remark this answers is not in the document]
Comment (Alberto-Codes) [reply .github/instructions/pytest.instructions.md]: Good catch — added a sentence to the convention: "For files using `pytest.importorskip()`, place `pytestmark` after the `importorskip` call and any conditional imports it gates — `importorskip` is treated as part of the import block." Addressed in latest push.
